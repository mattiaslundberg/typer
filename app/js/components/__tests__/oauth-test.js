import React from 'react'
import sd from 'skin-deep'
import OAuth from '../oauth'

describe('OAuth', () => {
  let render = () => {
    return sd.shallowRender(<OAuth />)
  }

  it('Render nothing before being loaded', () => {
    let tree = render()
    expect(tree.getRenderOutput()).toBe(null)
  })

  it('Render login link when received', () => {
    let tree = render()
    let instance = tree.getMountedInstance()
    instance.handleResponse(JSON.stringify({
      auth_url: "http://hello.com",
    }))

    expect(tree.getRenderOutput()).toMatchSnapshot()
  })

  it('Render logged in status when received', () => {
    let tree = render()
    let instance = tree.getMountedInstance()
    instance.handleResponse(JSON.stringify({
      auth_url: "http://hello.com",
      name: "Mattias",
      email: "mattias@example.com",
    }))

    expect(tree.getRenderOutput()).toMatchSnapshot()
  })
})
