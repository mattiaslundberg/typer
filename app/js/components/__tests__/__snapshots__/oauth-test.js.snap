exports[`OAuth Render logged in status when received 1`] = `
<div
  className="logout">
  Logged in as 
  Mattias
  .
  <a
    href="/api/logout/">
    Logout
  </a>
</div>
`;

exports[`OAuth Render login link when received 1`] = `
<div
  className="login">
  <a
    href="http://hello.com">
    Login using Google
  </a>
</div>
`;
